### PR TITLE
Add Travis CI setup

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
(Just copied from ocean.)

Based on #1.

### CODE REVIEW

**Comment by nemanja-boric-sociomantic** [comment]
We need to alert `.gitmodules` to use `https`.

**Comment by nemanja-boric-sociomantic** [comment]
And it looks like we need `Dockerfile.trusty` after all (it runs the trusty builds as well)

**Comment by gavin-norman-sociomantic** [comment]
Updated.

**Comment by nemanja-boric-sociomantic** [comment]
Ok, let's see if it gets green.

**Comment by nemanja-boric-sociomantic** [comment]
Bah, `docker/Dockerfile` is `docker/build` now: https://github.com/sociomantic-tsunami/ocean/blob/v2.x.x/docker/build and it's a script instead of actually being dockerfile.

**Comment by gavin-norman-sociomantic** [comment]
Ok, try again...

**Comment by nemanja-boric-sociomantic** [comment]
Missing `-x` permission on `docker/build`.

**Comment by gavin-norman-sociomantic** [comment]
Oops. Try again...

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        - DC=dmd1 DIST=xenial F=production AFTER_SCRIPT=1
+        - DC=dmd1 DIST=xenial F=devel
+        - DC=dmd-transitional DIST=trusty F=production
+        - DC=dmd-transitional DIST=trusty F=devel
+        - DC=dmd-transitional DIST=xenial F=production
+        - DC=dmd-transitional DIST=xenial F=devel
+
+install: beaver install
+
+script: BEAVER_DOCKER_VARS="DC" beaver run ci/travis.sh
```
Comment (nemanja-boric-sociomantic) [inline .travis.yml]: your script is now named `ci/run.sh`, you should change it here.

File: ci/after_success.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -xe
+
+test "$AFTER_SCRIPT" = "1" || exit 0
+
+# Submit code coverage report
+beaver run ci/codecov.sh
```
Comment (nemanja-boric-sociomantic) [inline ci/after_success.sh]: This file is not commited. perhaps we can just drop this for now (remove entire file and also remove after_success from `.travis.yml`

**Comment by gavin-norman-sociomantic** [comment]
Updated again.

**Comment by nemanja-boric-sociomantic** [comment]
One more time :-). All the `sh` needs `-x` it seems.

**Comment by gavin-norman-sociomantic** [comment]
Added `x` permission to `ci/run.sh`.

**Comment by nemanja-boric-sociomantic** [comment]
:green_heart:

**Comment by nemanja-boric-sociomantic** [comment]
@leandro-lucarella-sociomantic any last words?

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by gavin-norman-sociomantic** [comment]
#1 is merged. Unblocked.

**Comment by nemanja-boric-sociomantic** [comment]
I'll merge this to fix the v13.x.x branch, and @leandro-lucarella-sociomantic can followup here or in the next PRs, if there's something wrong.

**Pull request merged**